src/libraries/System.Net.Http.Json/tests/FunctionalTests/HttpClientJsonExtensionsTests.cs

```diff
@@ -0,0 +1,215 @@
+
+                        response = await client.PostAsJsonAsync(uri.ToString(), person, CancellationToken.None);
+                        Assert.True(response.StatusCode == HttpStatusCode.OK);
+
+                        response = await client.PostAsJsonAsync(uri, person, CancellationToken.None);
+                        Assert.True(response.StatusCode == HttpStatusCode.OK);
+                    }
+                },
+                async server => {
+                    for (int i = 0; i < NumRequests; i++)
+                    {
+                        HttpRequestData request = await server.HandleRequestAsync();
+                        ValidateRequest(request);
+                        Person per = JsonSerializer.Deserialize<Person>(request.Body, s_defaultSerializerOptions);
+                        per.Validate();
+                    }
+                });
+        }
+
+        [Fact]
+        public async Task TestPutAsJsonAsync()
+        {
+            const int NumRequests = 4;
+            await LoopbackServer.CreateClientAndServerAsync(
+                async uri =>
+                {
+                    using (HttpClient client = new HttpClient())
+                    {
+                        Person person = Person.Create();
+                        Type typePerson = typeof(Person);
+
+                        HttpResponseMessage response = await client.PutAsJsonAsync(uri.ToString(), person);
+                        Assert.True(response.StatusCode == HttpStatusCode.OK);
+
+                        response = await client.PutAsJsonAsync(uri, person);
+                        Assert.True(response.StatusCode == HttpStatusCode.OK);
+
+                        response = await client.PutAsJsonAsync(uri.ToString(), person, CancellationToken.None);
+                        Assert.True(response.StatusCode == HttpStatusCode.OK);
+
+                        response = await client.PutAsJsonAsync(uri, person, CancellationToken.None);
+                        Assert.True(response.StatusCode == HttpStatusCode.OK);
+                    }
+                },
+                async server => {
+                    for (int i = 0; i < NumRequests; i++)
+                    {
+                        HttpRequestData request = await server.HandleRequestAsync();
+                        ValidateRequest(request);
+                        Person obj = JsonSerializer.Deserialize<Person>(request.Body, s_defaultSerializerOptions);
+                        obj.Validate();
+                    }
+                });
+        }
+
+        [Fact]
+        public async Task TestHttpClientIsNullAsync()
+        {
+            HttpClient client = null;
+            string uriString = "http://example.com";
+            Uri uri = new Uri(uriString);
+
+            ArgumentNullException ex;
+            ex = await Assert.ThrowsAsync<ArgumentNullException>(() => client.GetFromJsonAsync(uriString, typeof(Person)));
+            Assert.Equal("client", ex.ParamName);
```

Comment:
A few things:
1. We expect these methods to throw these exceptions synchronously, right?  Assuming yes, we should be using Assert.Throws instead of Assert.ThrowsAsync.
2. The argument name can be tested as part of the same assert.
e.g.
```C#
AssertExtensions.Throws<ArgumentNullException>("client", () => { client.GetFromJsonAsync(uriString, typeof(Person)); });
```